### PR TITLE
Improve "add space" wizard

### LINKED ISSUE #10047: [Spaces] Add a single Space
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Userstory
As a regular user who was added to a Space, I want to quickly access the Spaces’ files via my Windows Explorer so that I can start working on those files in my familiar Windows-context.

### Step 1
- Existing Spaces show Space image (Square, cropped) or Space placeholder icon like: [remix icon "Function  fill"](https://github.com/Remix-Design/RemixIcon/blob/master/icons/System/function-fill.svg) or [material icon "Grid view"](https://fonts.gstatic.com/s/i/short-term/release/materialsymbolssharp/grid_view/fill1/48px.svg)
- Click on "Add a Space" opens Dialog

image

### Step 2
- Headline: "Add a Space"
- Copytext: "Select a Space to sync it your a local folder."
- Help-Link Label(+ Icons for External link): Learn more
- Help-Link Target: to be defined. If we don't have any further help documented, let's skip the Help-Link
- Spaces list shows only Spaces, that are not synced yet
- Only 1 Space can get selected
- Selection is indicated ex. via radio button
- Column "Space" shows Space `Image`/Placeholder, Space `Title`, Space `Subheadline` and weblink "Open in Web" incl. icon that indicates "external link"
- click on "Open in Web" opens the respective Space Frontpage in the browser
- Row-background could be slightly alternated so that rows can be distinguished better 
- Click on next: See following Scribble Nr. 3

image
image

### Step 3
- Headline: "Pick a local folder (optional)"
- Copytext: "Pick a local folder to synchronize the Space with."
- Help-Link Label(+ Icons for External link): Learn more
- Help-Link Target: to be defined. If we don't have any further help documented, let's skip the Help-Link
- Rename button label "Add Sync Connection" to "Save"

image

### Step 4
- New Space is shown as is already in testpilotcloud 3.0.0.xx

image

### Alternative solutions you considered

Tried to propose a not so expensive solution above and thus decided not to propose the following:
- The Expert dialog "Pick a local folder (optional)" should not be shown to every user (should only be shown on an Expert-path / on demand)
- Users should be able to add/remove multiple Spaces en-block ("Batch"). Dismissed this request as its a consequence of the mandatory "Pick a local folder (optional)" Dialog mentioned above.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by fmoc**
Follow-up for https://github.com/owncloud/client/pull/9575 and a few other PRs.

screenshot_2022-09-02_19-51-14

This PR rewrites a lot of spaces-related code, and should be reviewed carefully. The "open in browser" buttons are not working at the moment due to a server limitation (it does not send us URLs we could open in the browser as of yet). The placeholder is temporary, I hope in the future the server will just send us one.

Closes #10047.

### CODE REVIEW

**Review by erikjv — APPROVED** [review]

**Comment by fmoc** [comment]
Please re-review.

**Comment by fmoc** [comment]
screenshot_2022-09-06_09-24-33

screenshot_2022-09-06_09-24-39

File: src/gui/folderwizard/spacespage.cpp [hunk 1]
```diff
@@ -12,34 +12,52 @@
...
 
-    ui->widget->setAccount(acc);
+    connect(_ui->widget, &Spaces::SpacesBrowser::selectionChanged, this, &QWizardPage::completeChanged);
 
-    connect(ui->widget, &Spaces::SpacesBrowser::selectionChanged, this, &QWizardPage::completeChanged);
+    QTimer::singleShot(0, this, [this, accountPtr] {
+        auto drive = new OCC::GraphApi::DrivesJob(accountPtr);
+
+        connect(drive, &OCC::GraphApi::DrivesJob::finishedSignal, [drive, accountPtr, this] {
+            QList<Spaces::Space> spaces;
```
Comment (TheOneRing) [inline src/gui/folderwizard/spacespage.cpp:35]: spaces.reserve

File: src/gui/spaces/space.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+namespace {
+static const char contextC[] = "space";
+const auto personalC = QLatin1String("personal");
+const auto shareC = QLatin1String("virtual");
+}
+
+namespace OCC::Spaces {
+
+Space::Space(const QString &name, const QString &subtitle, const QUrl &webUrl, const QUrl &webDavUrl, const QUrl &imageUrl)
```
Comment (TheOneRing) [inline src/gui/spaces/space.cpp:29]: Any reason we need that constructor, can't we just own the Drive and operate on it?

File: src/gui/spaces/space.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+const QUrl &Space::imageUrl() const
+{
+    return _imageUrl;
+}
+
+Space Space::fromDrive(const OpenAPI::OAIDrive &drive)
+{
+    QUrl imageUrl = [special = drive.getSpecial()] {
+        const auto imageFolder = std::find_if(special.begin(), special.end(), [](const OpenAPI::OAIDriveItem &item) {
+            return item.getSpecialFolder().getName() == QStringLiteral("image");
```
Comment (TheOneRing) [inline src/gui/spaces/space.cpp:67]: At the moment thiss lambda is the only value added by all those lines of code.

**Review by TheOneRing — COMMENTED** [review]
Now that you removed the space model feel free to manually implement sorting, filtering and searhing.

File: src/gui/spaces/space.cpp (hunk 2 above)
Comment (fmoc) [reply src/gui/spaces/space.cpp:29]: It gives us some flexibility with regards to (future) testing.

**Comment by fmoc** [comment]
> Now that you removed the space model feel free to manually implement sorting, filtering and searching.

That's true, but I couldn't find a reasonable-effort way of rendering widgets with a delegate... Even when using rich text and falling back to some HTML links instead of buttons, some of the features provided by this PR would just not have worked without a custom delegate.

**Pull request merged**